File: LeetCode/Jump Game.cpp
class Solution {
public:
    bool canJump(int A[], int n) {
        int maxLength = 0;
        for (int i = 0; i < n; ++i)
            if (maxLength >= i)
                maxLength = max(maxLength, A[i] + i);

        if (maxLength >= n-1)
            return true;
        else
            return false;
    }
};

File: LeetCode/Jump Game.py
class Solution:
    # @param A, a list of integers
    # @return a boolean
    def canJump(self, A):
        if len(A)==1:
            return True

        f = [-1] * len(A)
        f[0] = A[0]
        for i in range(1, len(A)):
            if f[i-1] > 0:
                f[i] = max(A[i], f[i-1]-1)
            else:
                f[i] = -1
        # print f
        return f[len(A)-1] >= 0

if __name__ == '__main__':
    s = Solution()
    # A = [3,2,1,0,4]
    A = [2,3,1,1,4]
    print s.canJump(A)